### PR TITLE
Move (un)zip functions to archive.h and checks if (de)compression fails

### PR DESCRIPTION
**Pull request by autoantwort**
From the code review of  #307

### CODE REVIEW

File: src/vcpkg/archives.cpp [hunk 1]
```diff
@@ -181,4 +182,64 @@ namespace vcpkg
...
+
+#else
+        return cmd_execute_clean(Command{"zip"}
+                                     .string_arg("--quiet")
+                                     .string_arg("-y")
+                                     .string_arg("-r")
+                                     .path_arg(destination)
+                                     .string_arg("*")
+                                     .string_arg("--exclude")
+                                     .string_arg("\\*.DS_Store"),
```
Comment (strega-nil) [inline src/vcpkg/archives.cpp:206]: I do not like that we use `system` :|

Also, is there a reason this isn't just `.DS_Store`? My understanding is that `.DS_Store` will always be called exactly that.

**Review by strega-nil — APPROVED** [review]

File: src/vcpkg/archives.cpp [hunk 2]
```diff
@@ -181,4 +182,64 @@ namespace vcpkg
...
+
+    int compress_directory_to_zip(const VcpkgPaths& paths, const Path& source, const Path& destination)
+    {
+        auto& fs = paths.get_filesystem();
+        fs.remove(destination, VCPKG_LINE_INFO);
+#if defined(_WIN32)
+        auto&& seven_zip_exe = paths.get_tool_exe(Tools::SEVEN_ZIP);
+
+        return cmd_execute_and_capture_output(
+                   Command{seven_zip_exe}.string_arg("a").path_arg(destination).path_arg(source / "*"),
```
[the remark this answers is not in the document]
Comment (autoantwort) [reply src/vcpkg/archives.cpp:194]: Hm. Normally there are no `.DE_Store` files on windows

File: src/vcpkg/archives.cpp (hunk 1 above)
Comment (autoantwort) [reply src/vcpkg/archives.cpp]: I thought it was needed to match the file in sub dirs, but that was not needed

Comment (ras0219-msft) [reply src/vcpkg/archives.cpp]: I think we should escape `*` in `string_arg()` (on platforms where the shell will interpret it). If the intent was to actually pass `*`, it should be a `raw_arg()`.

**Review by ras0219-msft — APPROVED** [review]
This looks really good to me.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**